spec/lib/rack/content_type_default_spec.rb

```diff
@@ -15,63 +17,117 @@
-      Rack::ContentTypeDefault.new(App, :get).call(request1)
-      Rack::ContentTypeDefault.new(App, :get).call(request2)
-      request1['CONTENT_TYPE'].should == 'application/json'
-      request2['CONTENT_TYPE'].should_not == 'application/json'
+      Rack::ContentTypeDefault.new(App, :get).call(get_req)
+      Rack::ContentTypeDefault.new(App, :get).call(post_req)
+      get_req['CONTENT_TYPE'].should == 'application/json'
+      post_req['CONTENT_TYPE'].should_not == 'application/json'
     end
 
     it 'may be an array' do
-      request  = Rack::MockRequest.env_for("/api/v2/customers", method: :get)
-      Rack::ContentTypeDefault.new(App, [:get, :post]).call(request)
-      request['CONTENT_TYPE'].should == 'application/json'
+      Rack::ContentTypeDefault.new(App, [:get, :post]).call(get_req)
+      get_req['CONTENT_TYPE'].should == 'application/json'
     end
   end
 
   context 'content type' do
     it 'allows to specify the content type' do
-      request  = Rack::MockRequest.env_for("/api/v2/customers", method: :get)
-      Rack::ContentTypeDefault.new(App, :get, 'application/xml').call(request)
-      request['CONTENT_TYPE'].should == 'application/xml'
+      Rack::ContentTypeDefault.new(App, :get, 'application/xml').call(get_req)
+      get_req['CONTENT_TYPE'].should == 'application/xml'
     end
   end
-end
 
+  context 'paths' do
+    it 'sets the content type as appropriate if the path in the request matches one specified' do
+      get_req['PATH_INFO'] = 'api/v2/customers/2155/authenticate.json'
+      Rack::ContentTypeDefault.new(App, :get, 'application/xml', ['/show', '/authenticate.json']).call(get_req)
+      get_req['CONTENT_TYPE'].should == 'application/xml'
+    end
+
+    it 'does not set a content type if the path in the request does not match any specified' do
+      post_req['PATH_INFO'] = 'api/v2/customers/show'
+      Rack::ContentTypeDefault.new(App, [:get, :post], 'application/xml', ['/authenticate.json']).call(post_req)
+      post_req['CONTENT_TYPE'].should == nil
+    end
+
+    it 'sets content type as appropriate on all requests if paths specified as all' do
+      Rack::ContentTypeDefault.new(App, :get, 'application/xml', 'all').call(get_req)
+      get_req['CONTENT_TYPE'].should == 'application/xml'
+    end
+
+    it 'sets content type as appropriate on all requests if paths is none are explicitly specified' do
+      Rack::ContentTypeDefault.new(App, :post, 'application/xml').call(post_req)
+      post_req['CONTENT_TYPE'].should == 'application/xml'
+    end
+
+    it 'does not set content type on any requests if paths specified as empty ' do
+      get_req['PATH_INFO'] = 'api/v2/customers/2155/authenticate.json'
+      Rack::ContentTypeDefault.new(App, :get, 'application/xml', []).call(get_req)
+      get_req['CONTENT_TYPE'].should == nil
+    end
+  end
+
+  context 'default_based_on_path is set to true' do
+    it 'sets content type to application/json if path ends in .json' do
```

Comment:
how about `overrides the content type` rather than `sets` to emphasize that even i specified `application/xml` i would get `.json`.